Tests/RouterTests.swift

```diff
@@ -77,7 +92,35 @@ class RouterTests: QuickSpec {
             RouterTestServer.disable()
             Router.disableAll()
         }
-        
+
+        describe("Stop request loading") {
+            var router: Router!
+
+            beforeEach {
+                router = Router.register("http://www.test.com")
+                router.latency = 5 // very high latency to allow us to stop the request before execution
+            }
+
+            it("should not start a request, when the request loading is stopped") {
+
+                router.get("/users/:id") { request in
+                    XCTFail("Request should get stopped before and should not get executed")
+                    return nil
+                }
+
+                let requestURL = NSURL(string: "http://www.test.com/users/1")!
+                NSURLSession.sharedSession().dataTaskWithURL(requestURL) { (data, response, _) in
+                    // intentionally empty ... nothing to do here really
+                }
+
+                // We create a "dummy" NSURLProtocol object here, which is using the same URL (that we used to
+                // trigger the request). In order to check the request cancelation.
+                let urlRequest = NSURLRequest(URL: requestURL)
+                let urlProtocol = NSURLProtocol(request: urlRequest, cachedResponse: nil, client: nil)
+                router.stopLoading(urlProtocol)
+            }
+        }
+
```

Comment:
AFAIK you have to use XCTFail, anyway we need to give it the time to fail if the request is not correctly cancelled so we need to write the test without canceling implementation to check if they are working fine